src/ovl_1C_GrabBag/1837D0.c

```diff
@@ -1,17 +1,64 @@
 #include "common.h"
 
-INCLUDE_ASM(s32, "ovl_1C_GrabBag/1837D0", func_800F65E0_1837D0);
+#include "1837D0.h"
+
+
+void func_800F65E0_1837D0(void) {
+    func_80029090(0x10);
+    func_8002ADF0(&D_800EDEC0, 0x40);
+    func_8001DE70(0x10);
+    InitObjSystem(0x14, 0);
+    func_80060088();
+    func_8005D8B8(func_8005D384(0x7FDA, 0, 0, -1, func_8005EB1C), 0xA0);
+    func_8000942C();
+    D_800F6524.x = -10.0f;
+    D_800F6524.y = 0.0f;
+    D_800F6524.z = -105.0f;
+    D_800F33EC.y = -36.0f;
+    D_800F33EC.x = 0.0f;
+    D_800F33EC.z = 0.0f;
+    D_800EE98C = 1880.0f;
+    func_800F66F8_1838E8();
+    func_800F7260_184450();
+    func_800F67BC_1839AC();
+    func_800F6C08_183DF8();
+    func_800F7680_184870();
+    func_8002890C(0, 0, 0);
+    SetFadeInTypeAndTime(0, 0x10);
+}
+
+
+void func_800F66F8_1838E8(void) {
+    s16 temp_v0;
+    u8 temp_s1;
+
+    func_800178A0(1);
+    temp_v0 = func_800178E8();
+    temp_s1 = temp_v0 & 0xFF;
```

Comment:
Generally you wont have to manually & a variable like this unless it's flag/bits related. It should be possible to only use temp_s1 or temp_v0 for all of these function calls and then make sure the signatures of the functions are correct ( https://decomp.me/scratch/5zvoJ )